endpoints/citrus-http/src/test/java/org/citrusframework/http/message/HttpMessageBuilderTest.java

```diff
@@ -53,8 +55,10 @@ public void testDefaultMessageHeader() {
 
         //THEN
         assertEquals(builtMessage.getHeaders().entrySet().size(), 3);
-        assertEquals(message.getHeader(MessageHeaders.ID), builtMessage.getHeader(MessageHeaders.ID));
-        assertEquals(message.getHeader(MessageHeaders.TIMESTAMP), builtMessage.getHeader(MessageHeaders.TIMESTAMP));
+        assertNotNull(message.getHeader(MessageHeaders.ID));
+        assertNotNull(message.getHeader(MessageHeaders.TIMESTAMP));
+        assertNotEquals(message.getHeader(MessageHeaders.ID), builtMessage.getHeader(MessageHeaders.ID));
+        assertNotEquals(message.getHeader(MessageHeaders.TIMESTAMP), builtMessage.getHeader(MessageHeaders.TIMESTAMP));
```

Comment:
I think the negative case (`forceCitrusHeaderUpdate` being `false`) deserves a test too 😉

Reply:
I added respective test cases at DefaultMessage and HttpMessage level.